### ISSUE TITLE
Comments Loop Block: Tracking issue

### ISSUE THREAD

**Issue by SantosGuillamot** [WordPress/gutenberg#34994]
## Description

The idea of this issue is to end up with a Comments Loop Block (name may change) that loops over the comments of a given post. This would be similar to the [Query Loop](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query) used for posts. Users should be able to define and change the layout of the post comments directly from the Gutenberg editor, and both the editor and the frontend should match.

This is how the different comment blocks should look like:

![Comments Query Loop](https://user-images.githubusercontent.com/34552881/134122097-810be521-647b-4359-93fb-9ed4dc708731.png)

https://excalidraw.com/#json=6537760064143360,EgfZIljXt2SI2DTSFebHQQ

## Existing Solution / Workaround

Right now, there are two similar blocks:

- [Post Comments block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments): This block currently renders the `comments.php` file of the PHP theme in the frontend and a loop of paragraphs in the editor. This way, the editor and the frontend are different and users aren't able to change the layout unless they change the PHP code.
- [Post Comment block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment): This block allows users to insert a specific comment by ID in their posts/pages and edit the layout including other blocks like [Post Comment Author](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-author), [Post Comment Content](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-content) or [Post Comment Date](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-date). However, it isn't possible to loop through all the comments of a specific post.

## Tasks

These are the steps we should follow to finish the tracking issue:

*Please note that I'll be adding the relevant issues once they are created.*

- [x]  The Comments Query Loop block (https://github.com/WordPress/gutenberg/issues/34996). This should be similar to the [Query Loop](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query) used for posts.
- [x]  The Comments Template block (https://github.com/WordPress/gutenberg/issues/35000). This contains the block elements used to render a comment specified below, like the date, content or author. It should be similar to the [Post Template block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-template).
- [x]  Initial block elements for comments:
    - [x]  Comment Author (https://github.com/WordPress/gutenberg/issues/30572)
    - [x]  Comment Date (https://github.com/WordPress/gutenberg/issues/30573)
    - [x]  Comment Content (https://github.com/WordPress/gutenberg/issues/30574)
    - [x]  Comment Author Gravatar (https://github.com/WordPress/gutenberg/issues/30575)
    - [x]  Comment Reply Link (https://github.com/WordPress/gutenberg/issues/30576)
    - [x]  Edit Comment link (https://github.com/WordPress/gutenberg/issues/30577)
- [x]  Comments Pagination (https://github.com/WordPress/gutenberg/issues/35007). It should be similar to the [Query Pagination block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination).
    - [x]  Comments Pagination Next (https://github.com/WordPress/gutenberg/issues/35008). Similar to the [Query Pagination Next block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination-next).
    - [x]  Comments Pagination Previous (https://github.com/WordPress/gutenberg/issues/35009). Similar to the [Query Pagination Previous block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination-previous).
    - [x]  Comments Pagination Numbers (https://github.com/WordPress/gutenberg/issues/35010). Similar to the [Query Pagination Numbers block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination-numbers).

## Out of scope

There are some functionalities that could be nice to work on but that won't be covered in this tracking issue, although they may be addressed later in the future:

- Comments Form: There is an already working [Post Comments Form block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments-form), that allows users to leave a reply. Just adding the [Reply Link block](https://github.com/WordPress/gutenberg/issues/30576) pointing to this form should be enough for now.
- Comments Metadata: It could be nice to create a block to show any meta data added for the comments in the future.
- Comments Status: It could also make sense to create a block to show the status of the comments (approved, pending...), specially when they are submitted.

## Questions

- Once this is done, should we deprecate the existing [Post Comments block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments)?
- As explained previously, there are already working blocks for the [Post Comment Author](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-author), [Post Comment Date](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-date), [Post Comment Content](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-content) that are used for the [Post Comment Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment). I guess we could work on the existing ones right? Or would it be better to create new ones for this purpose?
- It seems to me that, while working on the Block Elements, we should keep in mind that they should also work fine for the [Post Comment Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment) right?
- Once the Comments Loop block is finished, should we even deprecate the [Post Comment Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment) and create a setting in the loop to select just an id?
- I feel that, as already pointed in other issues, it would be better to have two different blocks for the Comment Author and the Comment Author Avatar. However, right now in the posts, the Avatar is just a setting in the Post Author block. Is there any reason to not create two different blocks?
- Should we create in the future different block elements for the Comments Form block as well to be able to edit its layout?
- Anything else you think we should address and not covered here?

Of course, any feedback apart from these questions is more than welcome 🙂

## Previous work

There has been already some work done related to the Comments Block that was tracked in this issue → https://github.com/WordPress/gutenberg/issues/24101

**Comment by mcsf** [comment]
👋 Excited to see these great issues popping up!

> * Once this is done, should we deprecate the existing [Post Comments block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments)?

We can start with a soft deprecation, i.e. supporting the old block but deprioritising it (e.g. hiding it from the inserter). In other words, we don't need formal deprecation notices yet. The fact that it's a dynamic block means that, later on, we have some flexibility to determine how existing Post Comments blocks should be rendered.

> * As explained previously, there are already working blocks for the [Post Comment Author](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-author), [Post Comment Date](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-date), [Post Comment Content](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment-content) that are used for the [Post Comment Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment). I guess we could work on the existing ones right? Or would it be better to create new ones for this purpose?

If the old blocks' `usesContext: [ 'commentId' ]` directive is compatible with the new blocks' intended implementation, I'd say it's worth reusing.

> * It seems to me that, while working on the Block Elements, we should keep in mind that they should also work fine for the [Post Comment Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment) right?

Yes, as I mentioned above — unless supporting both becomes an unreasonable burden!

> * Once the Comments Loop block is finished, should we even deprecate the [Post Comment Block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment) and create a setting in the loop to select just an id?

IMO, it depends on what the benefits are, or what the use cases could be for this. It feels lower priority to me.

> * I feel that, as already pointed in other issues, it would be better to have two different blocks for the Comment Author and the Comment Author Avatar. However, right now in the posts, the Avatar is just a setting in the Post Author block. Is there any reason to not create two different blocks?

It's probably the trade-off between the flexibility of more granular blocks (Author + Avatar) vs. the user-friendliness and polish of dedicated multi-functional blocks (Author with integrated Avatar). With the ongoing improvements in layouts and alignments, the balance may soon tip towards granular blocks. I suggest experimenting with both and seeing what feels right. :)

**Comment by jameskoster** [comment]
@SantosGuillamot One design update we might consider: the new Dimensions and Layout panels will be useful on blocks like Avatar, but potentially all of them.

> With the ongoing improvements in layouts and alignments, the balance may soon tip towards granular blocks.

Probably worth getting a theme dev's input here (cc @kjellr), but I suspect that the flexibility afforded by separate blocks for author name and author avatar will be more useful in general.

**Comment by kjellr** [comment]
> Probably worth getting a theme dev's input here (cc @kjellr), but I suspect that the flexibility afforded by separate blocks for author name and author avatar will be more useful in general.

Yeah, the current iteration of the Post Author block is pretty inflexible and limiting for theme authors. There was a lot of talk about its issues (and the idea of splitting it into sub-blocks) in https://github.com/WordPress/gutenberg/issues/24952. This is one of the reasons the block was [left out of 5.8](https://github.com/WordPress/gutenberg/pull/31983).

**Comment by carolinan** [comment]
_Author with integrated Avatar_ -sounds like a good default block pattern or a block with inner blocks.
The post author block is really just waiting for a decision to be made, so that work can start.

**Comment by luisherranz** [comment]
@SantosGuillamot these three blocks don't have design, so I'm going to label them with `Needs Design`:
- https://github.com/WordPress/gutenberg/issues/35008
- https://github.com/WordPress/gutenberg/issues/35009
- https://github.com/WordPress/gutenberg/issues/35010

**Comment by SantosGuillamot** [comment]
> @SantosGuillamot One design update we might consider: the new Dimensions and Layout panels will be useful on blocks like Avatar, but potentially all of them.

I agree it makes sense for most of these blocks 🙂 . I can see that the Dimension panel was integrated in [this Pull Request](https://github.com/WordPress/gutenberg/pull/32392) and there is [another Pull Request](https://github.com/WordPress/gutenberg/issues/34698) to create a unified Layout panel. Do you know which is the best place to understand both panels better and how to integrate them in the blocks?

**Comment by jameskoster** [comment]
I believe the Group block has already implemented the Dimensions panel, so that might be a good place to look?

To clarify, it probably only makes sense for the _Layout_ panel to be added to container blocks in this context, IE the comment query loop and the comment template. But obviously we  can only do that once the feature has been added – my apologies, I thought it already had been.

**Comment by carolinan** [comment]
The post comment block is in the theme category, but the inner blocks are in the design category, is that correct or should they all be in the theme category?

**Comment by jameskoster** [comment]
Personally I think the inner blocks are more aligned with the other blocks in the theme category. 

But without wanting to open a can of worms, could there be an argument for a dedicated "Comments" section? There are quite a few blocks here...

**Comment by mtias** [comment]
As we work through these, I'd like to drop all the "Post" prefixes and just settle on "Comment Content", "Comment Author", etc.

**Comment by michalczaplinski** [comment]
Perhaps this is obvious but I'd like to clarify: 

If a user submits a new comment, we would still require a full page reload in order to show that new comment, correct? As opposed to creating a sort of "optimistic update" and showing the just-submitted comment without a reload.

**Comment by ntsekouras** [comment]
>If a user submits a new comment, we would still require a full page reload in order to show that new comment, correct? 

Do you mean like creating a comment in the front-end and show the results in the editor?

**Comment by luisherranz** [comment]
> If a user submits a new comment, we would still require a full page reload in order to show that new comment, correct?

That's correct Michal.

> Do you mean like creating a comment in the front-end and show the results in the editor?

I think he means rendering the comment in the browser after submission.

**Comment by michalczaplinski** [comment]
> I think he means rendering the comment in the browser after submission.

yup, that's what I meant. Thanks Luis!

**Comment by SantosGuillamot** [comment]
There has been some progress during the last weeks and I'd like to make a quick summary update. These are the issues we have worked on, and the status of each of them:

- [Comment Date block](https://github.com/WordPress/gutenberg/issues/30573): It has already been finished and merged! Thanks @carolinan  for the [Pull Request](https://github.com/WordPress/gutenberg/pull/35112).
- [Comment Author block](https://github.com/WordPress/gutenberg/issues/30572): @c4rl0sbr4v0 has been working on [this Pull Request](https://github.com/WordPress/gutenberg/pull/35595) which is almost finished. It's already in review and only some minor changes may be needed. You can check the video in the opening comment to check how it looks like.
- [Comment Avatar Author block](https://github.com/WordPress/gutenberg/issues/30575): @c4rl0sbr4v0 has been working [on this other Pull Request](https://github.com/WordPress/gutenberg/pull/35396) which is almost finished as well. There is a discussion about the width-height aspect ratio, but the code is almost finished.
- [Comment Content block](https://github.com/WordPress/gutenberg/issues/30574): @DAreRodz has been working on [this Pull Request](https://github.com/WordPress/gutenberg/pull/35183) and, again, it is almost finished. The code is already done and just a full code review is needed. Apart from that, there is [an open question](https://github.com/WordPress/gutenberg/pull/35183#issuecomment-944194359) to decide if we should include more typography options or not to keep consistency between blocks.
- [Comments Query Loop block](https://github.com/WordPress/gutenberg/issues/34996): This is probably the most complex issue and @michalczaplinski has been working on it in this [Pull Request](https://github.com/WordPress/gutenberg/pull/35231). There was some discussion to reduce the scope of the Pull Request, as it could get really long, and the first version should be ready soon.

Feel free to add anything that I'm missing, and you can find more information in each Pull Request 🙂

**Comment by SantosGuillamot** [comment]
A new update on the Post Comments blocks. This was the progress made during the last couple of weeks:

- [Comment Query Loop block](https://github.com/WordPress/gutenberg/issues/34996) & [Comment Template block](https://github.com/WordPress/gutenberg/issues/35000): These blocks are the wrappers for the rest of the block elements. An initial version of them was merged in this [Pull Request](https://github.com/WordPress/gutenberg/pull/35231) made by @michalczaplinski, and he is working on the next steps on [this other Pull Request](https://github.com/WordPress/gutenberg/pull/36065).
- [Comment Content block](https://github.com/WordPress/gutenberg/issues/30574): @DAreRodz  finished [this Pull Request](https://github.com/WordPress/gutenberg/pull/35183) and it has already been merged.
- [Comment Avatar Author block](https://github.com/WordPress/gutenberg/issues/30575): @c4rl0sbr4v0 finished [this Pull Request](https://github.com/WordPress/gutenberg/pull/35396) and it was merged as well.
- [Comment Reply Link block](https://github.com/WordPress/gutenberg/issues/30576): @DAreRodz worked on [this other Pull Request](https://github.com/WordPress/gutenberg/pull/35774) and it was merged too.
- [Comment Edit Comment Link block](https://github.com/WordPress/gutenberg/issues/30577): Another [Pull Request](https://github.com/WordPress/gutenberg/pull/35965) where @c4rl0sbr4v0  worked on and was merged.

For now, we will keep working on the aspects left in the Comment Query Loop block and start working on the blocks related to the Comments pagination. Feel free to add anything missing 🙂

**Comment by SantosGuillamot** [comment]
There has been some progress during the latest weeks regarding the Comments blocks:

- [Comment Query Loop block](https://github.com/WordPress/gutenberg/issues/34996) & [Comment Template block](https://github.com/WordPress/gutenberg/issues/35000): [@michalczaplinski](https://github.com/michalczaplinski) has been figuring how to handle nested comments and implementing it on [this Pull Request](https://github.com/WordPress/gutenberg/pull/36065). Moreover, [@c4rl0sbr4v0](https://github.com/c4rl0sbr4v0)  started [this other Pull Request](https://github.com/WordPress/gutenberg/pull/36623), which is needed to handle the pagination in inner blocks properly.
- [Comment Pagination Wrapper](https://github.com/WordPress/gutenberg/issues/35007): [@DAreRodz](https://github.com/DAreRodz) has been working on it on [this Pull Request](https://github.com/WordPress/gutenberg/pull/36577), which is currently in review.
- [Comment Pagination Next block](https://github.com/WordPress/gutenberg/issues/35008): [@c4rl0sbr4v0](https://github.com/c4rl0sbr4v0) started working on it on [this Pull Request](https://github.com/WordPress/gutenberg/pull/36562), but first it is needed to make the pagination work properly in the wrapper blocks. Once this is implemented, the [Comments Pagination Previous block](https://github.com/WordPress/gutenberg/issues/35009) should be pretty similar.

Apart from that, there is a discussion going on about if [multi-query comments loops should be provided](https://github.com/WordPress/gutenberg/issues/36642), the same as the Post Query Loop does. Any feedback/help is highly appreciated as this would probably affect the rest of the Pull Requests.

Feel free to add/correct anything 🙂

**Comment by SantosGuillamot** [comment]
There are now a few more pull requests opened related to the comments blocks which could make it difficult to keep track of the whole project. Trying to summarize it as much as possible, there is work being done in mainly two areas:

1. Wrapper blocks ([Comment Template Issue](https://github.com/WordPress/gutenberg/issues/35000) & [Comments Query Loop Issue](https://github.com/WordPress/gutenberg/issues/34996)).
2. Pagination.

In the first part, these are the **current open Pull Requests**:
- [Comment Template block: Handle nested comments](https://github.com/WordPress/gutenberg/pull/36065): This was the original PR to handle nested comments and it has already been merged. However, we still have to work on the following pull requests to consider it done.
- [Comment Template: Optimize the way pagination is handled for comments with replies](https://github.com/WordPress/gutenberg/issues/37153): This aims to solve an issue with the REST API not respecting the top-level comments setting, and it is difficult to replicate right now.
- [Comment Template: Consider using block placeholders when editing a template with no page context](https://github.com/WordPress/gutenberg/issues/37143).
- [Comment Template: Improve selection for the inner blocks](https://github.com/WordPress/gutenberg/issues/37154): This is a bug that currently affects the block editor experience.
- [Post Comments Loop: Unable to put comment blocks inside rows or columns](https://github.com/WordPress/gutenberg/issues/37181): This is another bug that needs to be addressed.

For the second part, the pagination, these are the **relevant Pull Requests**:
- [Add the Comments Pagination block](https://github.com/WordPress/gutenberg/pull/36872): This was already merged and it closed the [Comment Pagination wrapper issue](https://github.com/WordPress/gutenberg/issues/35007).
- [Comment Template: Add handling for the pagination inner blocks](https://github.com/WordPress/gutenberg/pull/36623): This Pull Request aims to make the Comments Query Loop able to read the pagination variables defined by Next/Previous/Number blocks or the URL, and display the comments consequently.
- [Comments Query Loop: Improve context handling in inner blocks](https://github.com/WordPress/gutenberg/pull/37196): This might replace the previous Pull Request.
- [Add the Comments Pagination Next block](https://github.com/WordPress/gutenberg/pull/36912) & [Add the Comments Pagination Previous block](https://github.com/WordPress/gutenberg/pull/36912): They are both advanced PRs and we will need to check how they integrate with the parent block after the rest of the PRs are merged.

Apart from that, there is an open conversation to [discuss if we should provide multi-query comments loops](https://github.com/WordPress/gutenberg/issues/36642).

As always, any feedback, clarification or correction is more than welcome 🙂

**Comment by SantosGuillamot** [comment]
All the blocks defined in the [opening comment](https://github.com/WordPress/gutenberg/issues/34994#issue-1002168008) are already merged 👏. 

Now, there is still some work going on on some tweaks and fixes to consider the project finished:
- [Add sorting by newer / older first options](https://github.com/WordPress/gutenberg/pull/37297): This is currently in review. It is being added an “Inherit from settings” option that would help to have it similar to your site options. That way it should be easier to fix some bugs.
- [Optimize the way pagination is handled for comments with replies](https://github.com/WordPress/gutenberg/issues/37153): This seems the most complicated PR left. It aims to solve an issue with the REST API not respecting the top-level comments setting, and it is difficult to replicate right now.
- [Consider using block placeholders when editing a template with no page context](https://github.com/WordPress/gutenberg/issues/37143): It doesn't seem an easy issue, cause there is [a similar one for Query Loop](https://github.com/WordPress/gutenberg/issues/26190). Having “real” data can cause performance issues as well as having no comments for making the templates if you are creating a new site.
- [Improve selection for the inner blocks](https://github.com/WordPress/gutenberg/issues/37154): This is a bug with the inner blocks.
- [Unable to put comment blocks inside rows or columns](https://github.com/WordPress/gutenberg/issues/37181): This is another bug, although there isn't a draft PR yet.
- [Link showing on the last page when selecting a custom "items per page" setting](https://github.com/WordPress/gutenberg/issues/37560): Another bug here. This will be addressed after landing [Add sorting by newer / older first option](https://github.com/WordPress/gutenberg/pull/37297).
- [Unify the markup on all pagination blocks rendering on PHP](https://github.com/WordPress/gutenberg/issues/37561): This shouldn't be a blocker, but it could be related with a few WordPress Core functions that make it a bit more difficult.

**Comment by SantosGuillamot** [comment]
This has been the progress in some of the issues/PRs mentioned in the previous comment:
- [Add sorting by newer / older first options](https://github.com/WordPress/gutenberg/pull/37297): @c4rl0sbr4v0 worked on this and it has already been merged.
- [Optimize the way pagination is handled for comments with replies](https://github.com/WordPress/gutenberg/issues/37153): @DAreRodz has started working on [this Pull Request](https://github.com/WordPress/gutenberg/pull/38187).
- [Consider using block placeholders when editing a template with no page context](https://github.com/WordPress/gutenberg/issues/37143): @c4rl0sbr4v0  has been working on a quick fix in [this Pull Request](https://github.com/WordPress/gutenberg/pull/38072). It currently needs some feedback to set a maximum number of comment placeholders.
- [Improve selection for the inner blocks](https://github.com/WordPress/gutenberg/issues/37154): @michalczaplinski has started working on this as well.

**Comment by SantosGuillamot** [comment]
A new update about the progress made:
- [Improve selection for the inner blocks](https://github.com/WordPress/gutenberg/issues/37154): @michalczaplinski handled it in [this Pull Request](https://github.com/WordPress/gutenberg/pull/38263) that has already been merged.
- [Consider using block placeholders when editing a template with no page context](https://github.com/WordPress/gutenberg/issues/37143): @c4rl0sbr4v0 has kept working on [this Pull Request](https://github.com/WordPress/gutenberg/pull/38072). It just needs the last review before being merged.
- [Optimize the way pagination is handled for comments with replies](https://github.com/WordPress/gutenberg/issues/37153): @DAreRodz has been working on [this Pull Request](https://github.com/WordPress/gutenberg/pull/38187). He recently posted a [detailed update](https://github.com/WordPress/gutenberg/pull/38187#issuecomment-1032536622) on it.
- [Try/unify user avatar](https://github.com/WordPress/gutenberg/pull/38591): @c4rl0sbr4v0 has started working on this Pull Request,  Proof of Concept of having a generic Avatar block for work in both comments and Post Author blocks.

**Comment by t-pabla** [comment]
Simple question: How would you remove (1) "date" , (2) "at", and (3) "time" from post comments?
WP 5.9
twentytwentytwo theme

**Comment by SantosGuillamot** [comment]
> Simple question: How would you remove (1) "date" , (2) "at", and (3) "time" from post comments?

These blocks haven't been included in WP 5.9, so you are probably using the [Post Comments block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments). This block currently renders the `comments.php` in the frontend, so you would need to change that. I haven't checked it but I assume you would need to do so via filters. Not sure if it is the best way, but making a quick search I found [this post](https://www.wpbeginner.com/wp-themes/how-to-remove-date-and-time-from-wordpress-comments/) that could be helpful.

In case you are using the Comments Loop block (this issue), just removing the [Comment date block](https://github.com/WordPress/gutenberg/issues/30573) should be enough.

**Comment by t-pabla** [comment]
Thanks for your reply , @SantosGuillamot 

In 5.9 editor, when I open single.php, I see "Post comments block". But I do NOT see "Comment date block" you referred to.

Where can this "Comment date block" be found? 

I use 5.9 and twentytwentytwo theme.

thanks again.

**Comment by SantosGuillamot** [comment]
### 

Sorry, I believe I didn’t explain myself correctly. The “Comments date block” hasn’t been included in WordPress 5.9 because the blocks mentioned in this particular issue aren’t finished yet.

You are probably using this other [Post Comments block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments), which calls the traditional function [`comment_template()`](https://developer.wordpress.org/reference/functions/comments_template/) and renders the `comments.php` file. So, if you want to remove the date, you would need to do something similar to [this post](https://www.wpbeginner.com/wp-themes/how-to-remove-date-and-time-from-wordpress-comments/), although I haven’t tested it. Hopefully, once these new blocks are released, it gets easier to make these changes.

**Comment by t-pabla** [comment]
Thanks Santos.

>>>>`The “Comments date block” hasn’t been included in WordPress 5.9 because the blocks mentioned in this particular issue aren’t finished yet.`

That's what I suspected.

Thanks again.

**Comment by SantosGuillamot** [comment]
I wanted to share a new update on the tracking issue and also what, in my opinion, is left to consider this project finished.

## Status of the issues/PRs mentioned in the last update

- [Consider using block placeholders when editing a template with no page context](https://github.com/WordPress/gutenberg/issues/37143): This issue has already been closed thanks to [this Pull Request](https://github.com/WordPress/gutenberg/pull/38072) @c4rl0sbr4v0  has been working on.
- [Optimize the way pagination is handled for comments with replies](https://github.com/WordPress/gutenberg/issues/37153): This one has been already solved as well in [this Pull Request.](https://github.com/WordPress/gutenberg/pull/38187) Thanks @DAreRodz  for working on it.
- [Try/unify user avatar](https://github.com/WordPress/gutenberg/pull/38591): @c4rl0sbr4v0  has been working more on this Pull Request.

## What is left to consider the tracking issue finished
There are a few more issues in the [GitHub project](https://github.com/WordPress/gutenberg/projects/61), but I believe not all of them have to be addressed right now to consider this tracking issue finished. I’ve tried to sum up which ones, **in my opinion**, are needed before closing the project:

- [Duplicated settings in inspector controls on Comments Query Loop block](https://github.com/WordPress/gutenberg/issues/38080).
- [Post Comments Loop: Unable to put comment blocks inside rows or columns](https://github.com/WordPress/gutenberg/issues/37181).
- [Discussion settings are not respected when using inherit option in Comments Query Loop block](https://github.com/WordPress/gutenberg/issues/39157).
- [Next Comments Block: Link showing on the last page when selecting a custom "items per page" setting](https://github.com/WordPress/gutenberg/issues/37560).
- [Try/unify user avatar](https://github.com/WordPress/gutenberg/pull/38591): This one is not directly related but it will affect the Comment Author Avatar.

As I've said, this is just my opinion, so any feedback is welcome 🙂

**Comment by SantosGuillamot** [comment]
New update on the issues that are left:
- [Discussion settings are not respected when using inherit option in Comments Query Loop block](https://github.com/WordPress/gutenberg/issues/39157): @c4rl0sbr4v0  worked on [this Pull Request](https://github.com/WordPress/gutenberg/pull/39204) that has already been merged.
- [Duplicated settings in inspector controls on Comments Query Loop block](https://github.com/WordPress/gutenberg/issues/38080): @michalczaplinski  has been working on [this Pull Request](https://github.com/WordPress/gutenberg/pull/39472) that will solve this issue. It is already ready for review.
- [Next Comments Block: Link showing on the last page when selecting a custom "items per page" setting](https://github.com/WordPress/gutenberg/issues/37560). There has been an interesting discussion in [this Pull Request](https://github.com/WordPress/gutenberg/pull/39227) and there is already a working solution. Only some e2e tests are needed to consider it complete.
- [Try/unify user avatar](https://github.com/WordPress/gutenberg/pull/38591): @c4rl0sbr4v0  has been working on it and it is already being reviewed.
- [Post Comments Loop: Unable to put comment blocks inside rows or columns](https://github.com/WordPress/gutenberg/issues/37181): This one seems a bit trickier. As discussed in the issue, @DAreRodz is researching how the `parent` property could look at all the ancestors, because it seems that would solve the issue.

Apart from that, this other issue was discovered, [Cannot select inner blocks in the Comments Query Loop block](https://github.com/WordPress/gutenberg/issues/39443), and it is already being fix in [this Pull Request](https://github.com/WordPress/gutenberg/pull/39472).

Moreover, as shared in [this Pull Request](https://github.com/WordPress/gutenberg/pull/39227), there are some issues if the Comments Query Loop settings aren’t inherited from the Discussion settings. This means that, initially, this option will be disabled until it is clearer how to approach it.

**Comment by mryudha** [comment]
Hi im new here on github. I don't know if its right to post my issue here or not. 

My issue is with comment pagination, i already open ticket on https://wordpress.org/support/topic/comment-pagination-got-mixed-when-i-try-option-break-comment-to-50-top-level/. Please check it. If im wrong to post here, do let me know the right way to post it.

Thanks.

**Comment by jasmussen** [comment]
I took a quick look at the inspector for the comments query loop:
<img width="680" alt="Comments Query Loop, before" src="https://user-images.githubusercontent.com/1204802/159663334-82f9b7e2-02be-4a04-a306-809267d20717.png">

I think we can do some small things here to refresh it a little bit:

<img width="688" alt="Comments Query Loop i2" src="https://user-images.githubusercontent.com/1204802/159663439-fd07abe4-118d-4b84-bf03-aab08e589ab2.png">

* I experimented with calling it just "Comments Loop". It seems fine to keep calling it "Comments Query Loop" if you prefer?
* A tweaked phrasing for the "Inherit" toggle, with separate help text that links to the WP-admin discussion settings page.
* I think we may have discussed this in one of the PRs, but is "Order" sufficient, or does it have to be "Order by"?

Some of the newer components that are shown are being developed separately, so that's not specifically an action item (the range slider for Items Per page), but I included the design targets in the mockups above for posterity! Is this helpful?

**Comment by michalczaplinski** [comment]
Thanks for the mockups @jasmussen ! 🙂 They definitely make for a better UI/UX!

That said, we have (for now) [removed the "inherit discussion settings" option](https://github.com/WordPress/gutenberg/pull/39664) altogether. The Comments Query Loop is now going to **always** inherit the settings from the core Discussion Settings.

The main reason for this change is explained in https://github.com/WordPress/gutenberg/pull/39227#issuecomment-1061996417 but, in short, the main reason was that when the core Discussion and block settings have different `perPage` values, the user will likely get redirected to an incorrect `/{post}/comment-page-X` page after having submitted a new comment.

So, unfortunately, at least for now, we're not going to implement those UI improvements until there is a clear path forward that includes fixing this problem. @SantosGuillamot is going to make a longer write-up (I think in a new issue) that includes more details so that further work can be picked up by any contributor.

**Comment by jasmussen** [comment]
> That said, we have (for now) https://github.com/WordPress/gutenberg/pull/39664 altogether. The Comments Query Loop is now going to always inherit the settings from the core Discussion Settings.

Even better!

**Comment by gziolo** [comment]
How does this functionality look in the terms of intent to ship in the upcoming WordPress 6.0 release? My understanding is that the development is almost finished and it's only a few final touches necessary. All blocks are still behind the experimental flag in the Gutenberg plugin:
https://github.com/WordPress/gutenberg/blob/1223f9309f48b3165427d1b3afbcc9f233de0e6f/packages/block-library/src/index.js#L262-L273

There is also an open question what should happen to the Post Comments block shipped with WordPress 5.9 which covers nearly the same functionality but with a completely different approach:
https://github.com/WordPress/gutenberg/blob/1223f9309f48b3165427d1b3afbcc9f233de0e6f/packages/block-library/src/index.js#L207

Beta 1 for WordPress 6.0 is planned for April 12th according to https://make.wordpress.org/core/6-0/. It means that Gutenberg v13.0 RC1 (planned for April 6th) is the last version where we can promote the Comments Query Loop block to shippable in WordPress 6.0.

**Comment by luisherranz** [comment]
> I experimented with calling it just "Comments Loop". It seems fine to keep calling it "Comments Query Loop" if you prefer?

@gziolo also mentioned dropping the "Query" word from the name [here](https://github.com/WordPress/gutenberg/issues/39648#issuecomment-1075993917).

If you think that it should be named "Comments Loop" even if the non-inherited options are added in the future (which is the "query" part), then I think we should do the renaming before this gets public in WP 6.0 👍

**Comment by jasmussen** [comment]
> If you think that it should be named "Comments Loop" even if the non-inherited options are added in the future (which is the "query" part), then I think we should do the renaming before this gets public in WP 6.0 👍

I don't have a strong opinion to be honest. Happy to defer!

**Comment by jameskoster** [comment]
If we're omitting custom queries, might we call it "Post Comments" to be consistent with other blocks like "Post Content", "Post Title", etc?

**Comment by luisherranz** [comment]
I don't have a strong opinion, but @mtias suggested the opposite in this comment: https://github.com/WordPress/gutenberg/issues/34994#issuecomment-929291106

**Comment by gziolo** [comment]
> If we're omitting custom queries, might we call it "Post Comments" to be consistent with other blocks like "Post Content", "Post Title", etc?

The challenge is that there is already "Post Comments" block in WordPress core (and Gutenberg plugin):

https://github.com/WordPress/wordpress-develop/blob/921d2805a0ebaf4db9f75ea9f5af46e48d6f68ac/src/wp-includes/blocks/post-comments/block.json#L4

That's is the biggest question what do we do about that legacy version that is a single block mostly based on PHP function calls and can't be customized in the UI.

**Comment by jameskoster** [comment]
> I don't have a strong opinion, but @mtias suggested the opposite in this comment: https://github.com/WordPress/gutenberg/issues/34994#issuecomment-929291106

I _think_ he was referring specifically to the children. I've suggested removing the Post prefix from all of these blocks before (IE "Post Date" becomes "Date") and there wasn't much appetite. 

> The challenge is that there is already "Post Comments" block in WordPress core (and Gutenberg plugin):

I don't know if it's possible, but shouldn't this block replace that one? It seems to serve the same purpose.

**Comment by cbravobernal** [comment]
The only issue is that the legacy version works fine with non-FSE themes, applying the styles defined on themes, using the default PHP functions. I guess we cannot replace it until we upgrade some WordPress versions.

The new `Comments` block still requires some configuration in order to have a nice design, now seems pretty raw when you insert it.

Is it possible to add some default styling? Should we create some patterns once we can use `Rows` and `Columns` inside `Comment Template`?

That way could be ready for 6.0 I guess.

**Comment by jameskoster** [comment]
I suppose the other problem is backwards compatibility for block themes that are using the existing Post Comments block in their templates.

**Comment by luisherranz** [comment]
> the other problem is backwards compatibility

I believe the pattern that has been used in situations like this is to hide the block (instead of deleting it), so it keeps working if it already exists, but new users/themes can't add it.

@gziolo: is that correct?

**Comment by gziolo** [comment]
> > the other problem is backwards compatibility
> 
> I believe the pattern that has been used in situations like this is to hide the block (instead of deleting it), so it keeps working if it already exists, but new users/themes can't add it.

Yes, there are a few blocks that no longer show up in the inserter with `supports.inserter` set to `false`:

https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#inserter

**Comment by SantosGuillamot** [comment]
After receiving a lot of great feedback in the [Help Test the Comments Blocks for WordPress 6.0 post](https://make.wordpress.org/test/2022/04/08/help-test-the-comments-blocks-for-wordpress-6-0/), there is some work needed to include the Comments blocks in WordPress 6.0. This is a summary of the issues/PRs that are currently considered (thanks @ockham for the help here 🙂 ):

### List of blockers
As Justin shared in this [comment](https://href.li/?https://make.wordpress.org/test/2022/04/08/help-test-the-comments-blocks-for-wordpress-6-0/#comment-2431) on the CfT, these three issues could be considered blockers:

- [x] Missing `#comment-id` on the comment wrapper element.
  - [x] Fixed by @c4rl0sbr4v0 and @ockham in [#40268](https://href.li/?https://github.com/WordPress/gutenberg/pull/40268).
  - [x] [Cherry-picked](https://github.com/WordPress/gutenberg/pull/40268#issuecomment-1102659742) to GB's `wp/6.0` branch
  - [x] [Backported to WP](https://github.com/WordPress/gutenberg/pull/40268#issuecomment-1102746535) per https://github.com/WordPress/wordpress-develop/pull/2603
- [x] Missing output of `comment_class()` on wrapper element (see post query implementation).
  - [x] Fixed by @michalczaplinski in [#40330](https://href.li/?https://github.com/WordPress/gutenberg/pull/40330).
  - [x] [Cherry-picked](https://github.com/WordPress/gutenberg/pull/40330#issuecomment-1102659570) to GB's `wp/6.0` branch
  - [x] [Backported to WP](https://github.com/WordPress/gutenberg/pull/40330#issuecomment-1103855023) per https://github.com/WordPress/wordpress-develop/pull/2603
- [ ] No comments title block (e.g., “X responses to post title”).
  - [x] Fixed by @c4rl0sbr4v0 in [#40419](https://href.li/?https://github.com/WordPress/gutenberg/pull/40419)
  - [ ] Needs cherry-picking
  - [ ] Needs backporting

As a follow-up to item 1, it could be considered:

- [x] Apply some finishing styling touches to the inlined Post Comments Form (Fixed by [#40359](https://href.li/?https://github.com/WordPress/gutenberg/pull/40359)).

Moreover, @peterwilsoncc also fixed:
- [x] Comment meta data isn’t queued for lazy loading. [#40242](https://github.com/WordPress/gutenberg/pull/40242)

### Stabilizing the Post Comments Form block.

It seems that the Post Comments Form should be added to the Comments Query Loop block’s default template ([#40256](https://href.li/?https://github.com/WordPress/gutenberg/pull/40256)), as people get confused by the missing comment submission form if they only insert the Comments Query Loop, which many may think of as a 1:1 replacement for the Post Comments block.

Unfortunately, this is blocked by the Post Comments Form currently being [experimental](https://href.li/?https://github.com/WordPress/gutenberg/blob/e853724475e5fced90e7b952ca465aeb54b85bf6/packages/block-library/src/index.js#L281). In order to stabilize it, it seems that at a minimum, this would be needed:

- [x] Change the Post Comment Form block’s visual representation in the Site Editor (GH issue with mockups: [#26999](https://href.li/?https://github.com/WordPress/gutenberg/issues/26999)). @michalczaplinski has been working on this Pull Request to solve it. [#40368](https://github.com/WordPress/gutenberg/pull/40368)
- [x] Removing the experimental status from the Post Comments Form block. (https://github.com/WordPress/gutenberg/pull/40368)

And finally:
- [x] Include Post Comments Form to the Comments Query Loop block’s default template ([#40256](https://href.li/?https://github.com/WordPress/gutenberg/pull/40256)).

### Retaining a fallback: The Post Comments block

It seems too late to include a [block transform from Post Comments to Post Query Loop](https://href.li/?https://github.com/WordPress/gutenberg/issues/40223). Since the Comments Query Loop block will have a few rough edges, it should be considered to keep the Post Comments block in the inserter (i.e. by removing [this line](https://href.li/?https://github.com/WordPress/gutenberg/blob/e853724475e5fced90e7b952ca465aeb54b85bf6/packages/block-library/src/post-comments/block.json#L37); no GH issue filed).

What do you think about this? Should we keep the current Post Comments block?

If so, although this block is already included in core, it may be needed to:

- [x] Implement a placeholder for the Site Editor (no GH issue filed), where it currently looks like this:
![post-comments-placeholder](https://user-images.githubusercontent.com/34552881/164000216-1ae297a2-343d-47cc-aeba-a0e0c0493e3a.png) 
   (fixed by https://github.com/WordPress/gutenberg/pull/40484)

Moreover, there is also this other PR:

- [x] Inline the comment form in place of the ‘Reply’ link the user clicked ([#40366](https://href.li/?https://github.com/WordPress/gutenberg/pull/40366)), much [like we did for the Comments Query Loop block](https://href.li/?https://github.com/WordPress/gutenberg/pull/40268).

### Renaming the Comments Query Loop block

@ockham has suggested that at this point – especially if the Post Comment Form is included, and potentially a Comments Title block – the block isn’t really a Comments Query Loop block, but more than that. As a user looking for “a block to display the comments”, they probably wouldn’t click on “Comments Query Loop” as their first choice in the inserter (especially if the Post Comments block is still available as a fallback).

So, what do you think about renaming the Comments Query Loop block to something that better captures what it is for, and what it does? Should it be called just “Comments”? Any other ideas?

---
Please, feel free to share anything else not considered here or any comments you have 🙂

**Comment by mtias** [comment]
Transforming the monolithic `Post Comments` block into a block made of blocks is the equivalent of the work that occurred on Galleries, Quote, etc. We should aim to replace the block as transparently as possible. For example, can we have a default template (inner blocks setup) that mimics the layout of the current post comments block? What if it just works as a fallback when no children are passed and when loaded in the editor you can pick a different pattern and that would transform into child blocks?

**Comment by ockham** [comment]
Thanks @mtias! A few thoughts (in different order):

> We should aim to replace the block as transparently as possible. For example, can we have a default template (inner blocks setup) that mimics the layout of the current post comments block?

We're kind of doing that already. This is what you get after inserting a Comments Query Loop block:

![image (1)](https://user-images.githubusercontent.com/96308/164236271-f5cbe524-c995-4256-be1c-ba1bbec0fa69.png)

> Transforming the monolithic `Post Comments` block into a block made of blocks is the equivalent of the work that occurred on Galleries, Quote, etc.

There's one difference: Post Comments is (fortunately!) a pretty basic _dynamic_ block, unlike those (rather complex) _static_ blocks you mentioned. So [while I first thought](https://github.com/WordPress/gutenberg/issues/40223#issuecomment-1095242667) that adding a transform would be too complex for now, maybe it could actually be pretty straight-forward 🤔  (@amustaque97 actually [started work on this already](https://github.com/WordPress/gutenberg/pull/40231) and later closed it after seeing my comment.)

> What if it just works as a fallback when no children are passed and when loaded in the editor you can pick a different pattern and that would transform into child blocks?

IIUC, the underlying assumption here is that Post Comments and Comments Query Loop are basically merged into one block, right? Well given the above, how would you feel about keeping them separate, with a block transform added?
(My major concern is that adding the UI for picking a pattern is more complex than a block transform and would make it more unlikely for the blocks to land in WP 6.0.)

**Comment by ockham** [comment]
Adding a list of PRs that we’ll also want to include in WP 6.0 (which aren’t in my post because I missed them, or they were filed later as follow-up bugfixes):

- #40284
- #40455
- #40471
- #40527

(I’ll update this comment with more PRs if needed.)

**Comment by ockham** [comment]
> For example, can we have a default template (inner blocks setup) that mimics the layout of the current post comments block? What if it just works as a fallback when no children are passed and when loaded in the editor you can pick a different pattern and that would transform into child blocks?

For reference, this is what inserting a (posts) Query Loop block looks like:

| With pattern | Start blank |
|-------------|-------------|
| ![post-query-loop](https://user-images.githubusercontent.com/96308/164468578-1f317eb8-9cde-4458-9e66-9823cf368568.gif) | ![post-query-loop-pattern](https://user-images.githubusercontent.com/96308/164468597-0d206cd9-8679-4a66-85f8-f64c95ae9e8a.gif) |

I imagine you're suggesting to do something like this for the Comments Query Loop block.

---

To me, what we have now -- which is basically

> a default template (inner blocks setup) that mimics the layout of the current post comments block

seems good as a baseline for WP 6.0. (Since users can customize those inner blocks by moving them around and/or styling them.) The pattern selection mechanism seems like a good feature to add later.

**Comment by ockham** [comment]
Update: Matías clarified via DM how he wants the Comments Query Loop logic to be merged into Post Comments, and how they should behave. I've filed #40521 with the details.

**Comment by ockham** [comment]
> Update: Matías clarified via DM how he wants the Comments Query Loop logic to be merged into Post Comments, and how they should behave. I've filed #40521 with the details.

Per [discussion](https://github.com/WordPress/gutenberg/issues/40521#issuecomment-1109589648) on the issue, and due to the [problems](https://github.com/WordPress/gutenberg/pull/40522#issuecomment-1109046302) we've encountered, we've decided not to attempt having the Comments Query Loop merged into the Post Comments block for WP 6.0. We'll revisit after it's released.

**Comment by ockham** [comment]
As a corollary to the [above](https://github.com/WordPress/gutenberg/issues/34994#issuecomment-1109592618), and per Slack discussion with @priethor:

The Post Comments block has been deprecated and removed from the inserter for a while, so people won’t be able to newly insert it. UX-wise, this is not so different from what the block merge would’ve achieved (where we would’ve retained the childless/monolithic version of the block for backwards compat, but upon newly inserting it, users would’ve gotten the modular version).

We still would like to [rename](https://href.li/?https://github.com/WordPress/gutenberg/pull/40506) Comments Query Loop (to Comments) though, as per our original plan.

**Comment by gziolo** [comment]
We caught two issues during backporting the latest changes to WordPress core:

1. A simple fix to sort out whitespaces to follow WP coding standards:
https://github.com/WordPress/gutenberg/blob/1e830c288322b3de97d679a1bf0ad34e2ef85f1a/packages/block-library/src/comments-title/index.php#L56-L68
2. The hook that is added globally in the Post Comments Form will also impact the Post Comments block:
https://github.com/WordPress/gutenberg/blob/1e830c288322b3de97d679a1bf0ad34e2ef85f1a/packages/block-library/src/post-comments-form/index.php#L58-L73
It should get applied in the render callback and removed after the output content is ready.

**Comment by gziolo** [comment]
Another small thing caught while testing in WordPress core.

Post Comments block on the front end:

<img width="639" alt="Screenshot 2022-04-26 at 15 09 54" src="https://user-images.githubusercontent.com/699132/165307292-35b3d545-44c6-47c3-b73c-9951469e715a.png">

Comments Query Loop on the front end:

<img width="635" alt="Screenshot 2022-04-26 at 15 09 48" src="https://user-images.githubusercontent.com/699132/165307395-16e4bbd6-66e3-4152-b8ab-95425d6efddf.png">

**Comment by cbravobernal** [comment]
Regarding the `title size`, is because we used h2 instead of h3. As @justintadlock recommended. We can go back and set the h3 as a default.
https://github.com/WordPress/gutenberg/pull/40419#issuecomment-1101769218

Regarding the `date`, we have a selector in the block that allows you to choose the date format. We can set with hours and minutes by default. Adding the 'at' would be trickier though. Also, the format is usually defined by the theme. In our block, we get the default format from the date settings.

Regarding the `Author`, we can address an issue to include a plaintext like in the Comments Title, so the user can write "says:", putting it as a default.

**Comment by gziolo** [comment]
Please note also "3 **R**esponses" vs "3 **r**esponses".

**Comment by ockham** [comment]
Thanks @gziolo! Working on fixing those here: #40628.

**Comment by peterwilsoncc** [comment]
As Beta 3 was the string freeze (no new strings), could the comment's title block use the same ones as in the template. This will avoid breaking the freeze & reduce the number of strings requiring translations.

```php
<?php
if ( 1 == get_comments_number() ) {
	printf(
		/* translators: %s: Post title. */
		__( 'One response to %s' ),
		'&#8220;' . get_the_title() . '&#8221;'
	);
} else {
	printf(
		/* translators: 1: Number of comments, 2: Post title. */
		_n( '%1$s response to %2$s', '%1$s responses to %2$s', get_comments_number() ),
		number_format_i18n( get_comments_number() ),
		'&#8220;' . get_the_title() . '&#8221;'
	);
}
?>
```
-- [source](https://github.com/WordPress/wordpress-develop/blob/6742d0d7a65e37f24c67eeaae373f4c086c277a5/src/wp-includes/theme-compat/comments.php#L34-L49)

I think the logic will need to be more complex to allow for users modifying the title in the block.

**Comment by cbravobernal** [comment]
> Please note also "3 Responses" vs "3 responses".

Now with the block, we are allowing some more options like:
- Show/Hide the comments count.
- Show/Hide the post title.
- Choose the text wanted for the string between the comments and the title.

Covering all those different options with different strings for uppercase/lowercase would be hard to maintain. I put it as an uppercase to cover when the Comments Count is hidden, but we can do it the other way. In both cases, the user will need to update the string if they change the default layout, according to their language.

Also, as @ockham mentioned [in the title PR](https://github.com/WordPress/gutenberg/pull/40419#issuecomment-1102968618), there is the possibility of having more than two different ways of plurals, so we should find a way to translate `_n` functionality into blocks.

**Comment by ockham** [comment]
> Please note also "3 **R**esponses" vs "3 **r**esponses".

This is being addressed by @c4rl0sbr4v0's https://github.com/WordPress/gutenberg/pull/40728.

**Comment by SantosGuillamot** [comment]
This is the progress of the latest Pull Requests. The changes made in the PRs that are backported will be included in 6.0 version. If I am not mistaken, the other ones will have to wait for future versions (6.0.1 is expected in June/July).

Feel free to correct me or add anything missing 🙂

## Merged and backported in beta/RC
- [Comments block: Fix glitches found while backporting](https://href.li/?https://github.com/WordPress/gutenberg/pull/40628).
- [Fix style comment awaiting moderation indentation](https://href.li/?https://github.com/WordPress/gutenberg/pull/40681).
- [Prevent comments rendering on password protected posts](https://href.li/?https://github.com/WordPress/gutenberg/pull/40755).
- [Prevent spam links of author’s name in pending moderation comments (take two)](https://href.li/?https://github.com/WordPress/gutenberg/pull/40702).
- [Comment Template: With pagination, make sure to request page 1 if there are no comments](https://href.li/?https://github.com/WordPress/gutenberg/pull/40759).

## Merged but NOT backported
- [Refactor: Reuse code of the editor placeholder across Post Comments and Post Comments Form](https://href.li/?https://github.com/WordPress/gutenberg/pull/40560).
- [Comments Title: Count toggle working in ‘Singular’ editing mode (fix)](https://href.li/?https://github.com/WordPress/gutenberg/pull/40728).
- [Comments Query Loop block: Rename to Comments (but keep ID)](https://href.li/?https://github.com/WordPress/gutenberg/pull/40763).

## Not finished yet
- [Comments block: Change ID to core/comments](https://href.li/?https://github.com/WordPress/gutenberg/pull/40506): We divided the rename of the comments into two PRs to prevent risks and include the [easiest one](https://href.li/?https://github.com/WordPress/gutenberg/pull/40763) in 6.0 (although it wasn’t finally included).
- [Block styles affect the message inside the Warning component](https://href.li/?https://github.com/WordPress/gutenberg/issues/40536).
- Post Comments placeholder: Fix left/right alignment of the “Older Comments” and “Newer Comments” pagination links in the Site Editor. (https://github.com/WordPress/gutenberg/issues/40827)
- Remove whitespace on top of Post Comments block Warning. (https://github.com/WordPress/gutenberg/issues/40826)
- Fix some translation issues in the Comments Title block and the Post Comments placeholder. (No issue filed yet. PR for Comments Title: https://github.com/WordPress/gutenberg/pull/40817)
- Decide if we are going to hide the Comments from the Post Editor as discussed [here](https://href.li/?https://github.com/WordPress/gutenberg/issues/40617#issuecomment-1110040233).

**Comment by ockham** [comment]
@SantosGuillamot I updated your [comment](https://github.com/WordPress/gutenberg/issues/34994#issuecomment-1117376308) to add two links to issues that I created (plus one for a PR).

**Comment by annezazu** [comment]
>Anything else you think we should address and not covered here?

In case it's helpful, some feedback came in through the FSE Outreach Program for the fourteenth call for testing that explored these new blocks. [Specifically, the following ideas were thrown out:](https://make.wordpress.org/test/2022/04/28/fse-program-testing-call-14-rallying-recipe-reviewers/#comment-2549)

- A block that shows when the last comment happened
- A block that shows how many people have commented total (across the whole site rather than just the post)

Happy to open individual issues for these if it's desired but wanted to lightly put on the radar for now as possible ideas to consider.

**Comment by cbravobernal** [comment]
Thanks for the feedback @annezazu 🎉 😄 

> A block that shows how many people have commented total (across the whole site rather than just the post)

There is already a Post Comments Count block, so we may just add an option to display the whole site comments count. It could be a `Good First Issue` if the community wants this feature. But I don't know if this would be a good Core block feature.

> A block that shows when the last comment happened

Would be this one just a date? I cannot find any use case here, but, with Latest Comments block, you can achieve something similar.

<img width="1724" alt="Screenshot 2022-05-25 at 11 53 19" src="https://user-images.githubusercontent.com/37012961/170235231-624b64fa-35a4-46f2-a61c-ee06c246dc44.png">

**Comment by annezazu** [comment]
> Would be this one just a date? I cannot find any use case here, but, with Latest Comments block, you can achieve something similar.

Yes! Just the date as a way to show perhaps more active commenting areas compared to others. Neat to see you can do this already with latest comments block 💥

**Comment by ockham** [comment]
Hey @annezazu , thanks a lot for the feedback! 😄 

> [Specifically, the following ideas were thrown out:](https://make.wordpress.org/test/2022/04/28/fse-program-testing-call-14-rallying-recipe-reviewers/#comment-2549)
>
> - A block that shows when the last comment happened
> - A block that shows how many people have commented total (across the whole site rather than just the post)

While those make sense, I’m afraid the team won’t prioritize these blocks in the near future. TBH, since their use cases seem a bit less common than our garden variety Comments blocks, I think it’s arguable that they could be third-party plugin material — especially the latter; and for the former, there’s the Latest Comments block that @c4rl0sbr4v0 pointed out that could serve as a substitute. If you think it would still make sense to have issues for those in the GB repo (potentially to discuss that these blocks _should_ be in GB/Core), please go ahead and file them 😊🙏 

---

FWIW, we’re going to close this tracking issue soon (since WP 6.0 has been released), and will file a new one with a list of some smaller follow-up fixes for WP 6.0.1 (but we won’t be introducing any new blocks there). Beyond that, the @WordPress/frontend-dx team will be focusing on Block Frontend Hydration, and on modularizing the Post Comments Form block.

**Comment by ZebulanStanphill** [comment]
> * A block that shows when the last comment happened
> 
> * A block that shows how many people have commented total (across the whole site rather than just the post)

These sound like ideas more suited for the dynamic tokens concept discussed in #39831, rather than blocks.

**Comment by SantosGuillamot** [comment]
Closing this in favor of #41451
